File: angular/moth/src/app/model/task.ts
export class Task {

  id: string;
  text: string;
  isDone: boolean;

  constructor(id?: string, text?: string, isDone?: boolean) {
    this.id = id;
    this.text = text;
    this.isDone = isDone;
  }

  static fromObject(obj: Object): Task {
    let task = new Task();
    task = Object.assign(task, obj);
    return task;
  }

  storable(): Object {
    const exportable = Object.assign({}, this);
    delete exportable.id;
    return exportable;
  }

}

File: flutter/moth/lib/models/task.dart
class Task {
  String text;
  bool isDone;
  String id;

  Task({this.text, this.isDone, this.id});

  Task.fromMap(Map<String, dynamic> data, String id) {
    this.id = id;
    this.text = data['text'];
    this.isDone = data['isDone'];
  }

  Map<String, dynamic> toMap() {
    return {
      "text": text,
      "isDone": isDone,
    };
  }
}
